nck/entrypoint.py

```diff
@@ -41,17 +41,17 @@ def build_commands(cli, available_commands):
 
 @cli.resultcallback()
 def process_command_pipeline(provided_commands, normalize_keys):
```

Comment:
Thank you for spotting the issue @tom-grivaud. Actually this is me who implemented this specific behavior when refactoring the code (thinking that it would make `nck/entrypoint.py` more readable), but I didn't think of this downside. We could change it for the below snippet (which is actually what was implemented before):

```
@cli.resultcallback()
def process_command_pipeline(provided_commands, normalize_keys):
    cmd_instances = [cmd() for cmd in provided_commands]
    provided_readers = list(filter(lambda o: isinstance(o, Reader), cmd_instances))	
    provided_writers = list(filter(lambda o: isinstance(o, Writer), cmd_instances))
    
    reader = provided_readers[0]
    for stream in reader.read():
        for writer in provided_writers:
            if normalize_keys and issubclass(stream.__class__, JSONStream):
                writer.write(NormalizedJSONStream.create_from_stream(stream))
            else:
                writer.write(stream)
```

Reply:
Yep that works better indeed. Can I push this snippet on this PR?

Reply:
yes